### ISSUE TITLE
Major update proposal

### ISSUE THREAD

**Issue by brodybits** [apache/cordova-node-xcode#20]
* drop support for deprecated Node.js versions 0.8, 0.10, 0.12, and 4
* simple-plist@0.3 update
* use ES2015 Object.assign as proposed in PR #14

**Comment by n1ru4l** [comment]
`Object.assign` is available since `4.9.1` and most projects that are using `cordova-node-xcode` require the minimum node version6/8 as a dependency.

Also node versions prior to `6` are afaik no longer maintained (https://github.com/nodejs/Release).

I think we should certainly start modernizing this code!

**Comment by brodybits** [comment]
I would like to get PR #24 and ideally PR #12 finished, to be published in a `1.0.1` release before starting the major updates. @n1ru4l I think it would be helpful if you could take a look at PR #12, to help determine whether or not we should merge it before publishing the next release.

**Comment by n1ru4l** [comment]
Also since nodeunit seems deprecated (https://www.npmjs.com/package/nodeunit#deprecated-project) and from my perspective the current tests are a mess. E.g. adding my test for (https://github.com/apache/cordova-node-xcode/pull/24) was frustrating since there was no nice way to see the actual differences of expected and actual for long strings.

Therefore I would recommend to update to a more mature testing framework. I personally would prefer jest (https://jestjs.io/).

Furthermore tools like prettier and eslint could furthermore improve the overall code quality/consistency. I have set those up for multiple projects and could do so for this one too!

**Comment by brodybits** [comment]
> Therefore I would recommend to update to a more mature testing framework.

Agreed; I would rather discuss this in a separate issue.

I think it would make sense but not be absolutely necessary to do this after we drop deprecated Node.js version.

**Comment by fbartho** [comment]
Could we possibly either convert this to TypeScript or start shipping TypeScript Definitions?

This would be super straightforward with the move to ES6 classes #38, as it would be nearly self-documenting!

**Comment by Esemesek** [comment]
We are converting https://github.com/react-native-community/cli to TypeScript and it would be awesome if this package had some kind of TS types: shipped with the package or hosted in DefinitelyTyped repository.

**Comment by erisu** [comment]
Closing as resolved and stale. Most of the items have been completed since the ticket was created, and there appears to be another ticket covering ES6.
